ui/src/app/views/workspace/component.ts

```diff
@@ -192,41 +218,52 @@ export class WorkspaceComponent implements OnInit, OnDestroy {
     });
   }
 
-  pollCluster(): Observable<Cluster> {
-    // Polls for cluster startup every minute.
+  private pollCluster(): Observable<Cluster> {
+    // Polls for cluster startup every 10s.
     return new Observable<Cluster>(observer => {
-      this.clusterService.getCluster(this.workspace.namespace, this.workspace.id)
-          .subscribe((cluster) => {
-        if (cluster.status !== 'Running' && cluster.status !== 'Deleting') {
-          setTimeout(() => {
+      this.clusterService.listClusters()
+        .subscribe((resp) => {
+          const cluster = resp.defaultCluster;
+          if (cluster.status !== ClusterStatus.RUNNING) {
+            setTimeout(() => {
               this.pollCluster().subscribe(newCluster => {
                 this.cluster = newCluster;
                 observer.next(newCluster);
                 observer.complete();
               });
-            }, 60000
-          );
-        } else {
-          this.cluster = cluster;
-          observer.next(cluster);
-          observer.complete();
-        }
+            }, 10000);
+          } else {
+            observer.next(cluster);
+            observer.complete();
+          }
       });
     });
   }
 
   cancelCluster(): void {
+    this.launchedNotebookName = '';
     this.clusterPulled = false;
   }
 
-  openCluster(): void {
+  openCluster(notebookName?: string): void {
     // TODO (blrubenstein): Make this configurable by environment
```

Comment:
Is it configurable by environment now? I think we're just using prod for everything, so this comment can go away?

Reply:
It's not currently configurable; done for now (though it's possible we might want to make this configurable in the future). I added a new TODO to instead generate the TS lib for the notebooks backend and use it here. This would give us direct access to the notebook.yaml host URL as well as the method bindings, which would match how the backend chooses which host to talk to.